### PR TITLE
Db implementation

### PR DESCRIPTION
**Pull request by Nashatyrev**
This is ongoing PR for db

### CODE REVIEW

File: db/core/src/main/java/org/ethereum/beacon/db/source/HoleyList.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+   * Returns element at index <code>idx</code>
+   * Empty instance is returned if no element with this index
+   */
+  Optional<V> get(long idx);
+
+  /**
+   * Puts element with index <code>size()</code>
+   */
+  default void add(V value) {
+    put(size(), value);
```
Comment (zilm13) [inline db/core/src/main/java/org/ethereum/beacon/db/source/HoleyList.java]: I'd return result here

File: build.gradle [hunk 2]
```diff
@@ -14,6 +14,8 @@ allprojects {
   targetCompatibility = '1.8'
 
   dependencies {
+    implementation 'com.google.code.findbugs:jsr305'
```
Comment (zilm13) [inline build.gradle]: What is it for?

File: chain/src/main/java/org/ethereum/beacon/chain/storage/AbstractChainStorage.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package org.ethereum.beacon.chain.storage;
+
+import org.ethereum.beacon.chain.storage.BeaconBlockStorage;
+import org.ethereum.beacon.chain.storage.BeaconStateStorage;
+import org.ethereum.beacon.chain.storage.BeaconTupleStorage;
+import org.ethereum.beacon.chain.storage.ChainStorage;
+import org.ethereum.beacon.chain.storage.impl.BeaconTupleStorageImpl;
+import org.ethereum.beacon.db.Database;
+
+public abstract class AbstractChainStorage implements ChainStorage {
```
Comment (zilm13) [inline chain/src/main/java/org/ethereum/beacon/chain/storage/AbstractChainStorage.java]: Doesn't look like something abstract.
Though I'm ok to keep naming simple if it's in specific package.

File: chain/src/main/java/org/ethereum/beacon/chain/storage/ChainStorage.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package org.ethereum.beacon.chain.storage;
+
+import org.ethereum.beacon.db.Database;
+
+/**
+ * Created by Anton Nashatyrev on 27.12.2018.
+ */
+public interface ChainStorage {
```
Comment (zilm13) [inline chain/src/main/java/org/ethereum/beacon/chain/storage/ChainStorage.java]: Doesn't look like something abstract.
Though I'm ok to keep naming simple if it's in specific package.

File: db/core/src/main/java/org/ethereum/beacon/db/source/HoleyList.java (hunk 1 above)
Comment (Nashatyrev) [reply db/core/src/main/java/org/ethereum/beacon/db/source/HoleyList.java]: What result?

File: build.gradle (hunk 2 above)
Comment (Nashatyrev) [reply build.gradle]: `@Nonnull`, `@Nullable`

File: chain/src/main/java/org/ethereum/beacon/chain/storage/ChainStorage.java (hunk 4 above)
Comment (Nashatyrev) [reply chain/src/main/java/org/ethereum/beacon/chain/storage/ChainStorage.java]: Not sure what you mean here... Let's talk on tomorrow call

File: build.gradle (hunk 2 above)
Comment (zilm13) [reply build.gradle]: Aaah, ok

File: db/core/src/main/java/org/ethereum/beacon/db/source/HoleyList.java (hunk 1 above)
Comment (zilm13) [reply db/core/src/main/java/org/ethereum/beacon/db/source/HoleyList.java]: Could it update old one, or it's always new?

File: chain/src/main/java/org/ethereum/beacon/chain/storage/ChainStorage.java (hunk 4 above)
Comment (zilm13) [reply chain/src/main/java/org/ethereum/beacon/chain/storage/ChainStorage.java]: I mean `ChainStorage` sounds abstract but `BeaconBlockStorage` etc are very specific.  So smth like `BeaconChainStorage` and `AbstractBeaconChainStorage` could be better.
But anyway I'm ok to keep it shorter like it is now.

File: db/core/src/main/java/org/ethereum/beacon/db/source/HoleyList.java (hunk 1 above)
Comment (Nashatyrev) [reply db/core/src/main/java/org/ethereum/beacon/db/source/HoleyList.java]: Yep, it's always new

**Pull request merged**